### PR TITLE
Add support for Quarto integration

### PR DESCRIPTION
**Pull request by wch**
This PR adds support for using Shiny in a Quarto document.

It also requires https://github.com/posit-dev/py-htmltools/pull/61

### CODE REVIEW

File: shiny/_app.py [hunk 1]
```diff
@@ -370,9 +384,30 @@ def _render_page(self, ui: Tag | TagList, lib_prefix: str) -> RenderedHTML:
...
+        doc = HTMLTextDocument(
+            page_html,
+            deps=[require_deps(), jquery_deps(), shiny_deps()],
+            deps_replace_pattern='<meta name="shiny-dependency-placeholder" content="">',
+        )
+
+        rendered = doc.render(lib_prefix=lib_prefix)
+        self._ensure_web_dependencies(rendered["dependencies"])
+
+        # TODO: scan for inlined <script> deps, inject them as well.
```
Comment (jcheng5) [inline shiny/_app.py:400]:
```suggestion
```

File: shiny/_main.py [hunk 2]
```diff
@@ -480,7 +500,29 @@ def static_assets(command: str) -> None:
...
+    type=str,
+)
+@click.argument(
+    "py_file",
+    type=str,
+    default=None,
+)
+def cells_to_app(json_file: str, py_file: str | None) -> None:
+    if py_file is None:
+        py_file = str(Path(json_file).parent / "app.py")
```
Comment (jcheng5) [inline shiny/_main.py:515]: app.py shouldn't be the default filename

Comment (jcheng5) [reply shiny/_main.py:515]: Or don't have a default

File: shiny/_app.py [hunk 3]
```diff
@@ -89,7 +97,7 @@ def server(input: Inputs, output: Outputs, session: Session):
 
     def __init__(
         self,
-        ui: Tag | TagList | Callable[[Request], Tag | TagList],
+        ui: Tag | TagList | Callable[[Request], Tag | TagList] | Path,
```
Comment (wch) [inline shiny/_app.py:100]: @jcheng5 What do you think about taking a `Path` object to indicate that it should read the UI from a file?

**Pull request merged**